### ISSUE TITLE
An option to turn off `nl2br()` for custom MyCodes

### ISSUE THREAD

**Issue by Ben-MyBB** [mybb/mybb#4226]
Some custom MyCodes like printing HTML codes using &lt;pre&gt; may have the issue of outputting &lt;br /&gt;for raw line breaks.

There are several threads demanding an option to disable nl2br() function in MyCode parsing:
https://community.mybb.com/thread-201870.html
https://community.mybb.com/thread-203937.html
https://community.mybb.com/thread-205148.html
https://community.mybb.com/thread-230056.html

Original thread: [An option to turn off `nl2br()` for custom MyCodes](https://community.mybb.com/thread-230595.html)

**Comment by Eldenroot** [comment]
I think this should be a plugin (plugin is already available). Do we need to add this into core? If yes, no problem, but this is quite specific.

**Comment by Sama34** [comment]
This seems like a bug to me, any parser option should be possible to be switched on or off, also, I think administrators should be able to create custom MyCodes similar to the _code_ or _php_ tags (in that nothing gets parsed within them nor breaks added).

I think the second part of my suggestion should be left for 1.9, whereas the first part of my comment should probably be in 1.8 but I won't be sure what others think before creating a PR.

**Comment by lairdshaw** [comment]
@Sama34, isn't the first part of your suggestion already met? It seems that it is already possible to supply to the parser a boolean value for the 'nl2br' option, which is respected here:

https://github.com/mybb/mybb/blob/0c868f1830bdbc56e95fd669cd8ff17c44422807/inc/class_parser.php#L243

Agreed that the second part should be deferred to 1.9.

**Comment by Sama34** [comment]
You are correct, I got my mind mixed there, I think this should be postponed for 1.9 to implement the second part, which should _fix_ the issues from the links in the OP.
